AK Zeus/app/src/main/java/ayds/zeus/songinfo/moredetails/view/MoreDetailsView.kt

```diff
@@ -57,39 +55,31 @@ class OtherInfoActivity : AppCompatActivity(), MoreDetailsView {
 
     private fun initObservers() {
         moreDetailsModel.cardObservable()
-                .subscribe { value -> updateWithNewCard(value) }
-    }
-
-    private fun updateWithNewCard(card: Card){
-        updateUiState(card)
-        showCardInfoActivity()
-    }
-
-    private fun updateUiState(card: Card) {
-        when (card) {
-            is EmptyCard -> updateNoResultUiState()
-            else -> updateCardUiState(card)
+            .subscribe { value ->
+                run {
+                    initSpinner(value)
+                    updateUiState(value)
+                }
+            }
+    }
+
+    private fun initSpinner(list: List<Card>) {
+        val spinnerList: MutableList<String> = mutableListOf()
+        for (card in list) {
+            spinnerList.add(card.source.sourceName)
+        }
```

Comment:
Se puede simplificar
```suggestion
        spinnerList.addAll(list.map { it.source.sourceName })
```